source/cloudserver/webapi/XPowerApi/XPowerApi/Controllers/UserCreationController.cs

```diff
@@ -0,0 +1,75 @@
+﻿using XPowerApi.Interfaces;
+using XPowerApi.Models.UserModels;
+using XPowerApi.Models.HomeModels;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+
+
+namespace XPowerApi.Controllers
+{
+    [Route("api/user")]
+    [ApiController]
+    public class UserCreationController : Controller
+    {
+        private readonly ILogger<UserCreationController> _logger;
+        private readonly IUserManager _userManager;
+        private readonly IHomeManager _homeManager;
+        private readonly ITokenManager<UserToken> _tokenManager;
+
+        public UserCreationController(
+            ILogger<UserCreationController> logger,
+            IUserManager userManager,
+            IHomeManager homeManager,
+            ITokenManager<UserToken> tokenManager
+        )
+        {
+            _logger = logger;
+            _userManager = userManager;
+            _homeManager = homeManager;
+            _tokenManager = tokenManager;
+        }
+
+        /// <summary>
+        /// Create user
+        /// </summary>
+        [AllowAnonymous]
+        [HttpPost]
+        [Route("CreateUser")]
+        public async Task<ActionResult<UserToken>> CreateUser(UserCreate userCreateInfo)
+        {
+            if (userCreateInfo == null)
+                return BadRequest(new { Message = "Missing parameter" });
+
+            //checking if username and password is not null or empty
+            if (
+                string.IsNullOrEmpty(userCreateInfo.UserName) ||
+                string.IsNullOrEmpty(userCreateInfo.Password) ||
+                string.IsNullOrEmpty(userCreateInfo.Email) ||
+                string.IsNullOrEmpty(userCreateInfo.FirstName) ||
+                string.IsNullOrEmpty(userCreateInfo.LastName)
+            )
+                return BadRequest(new { Message = "Username or Password is missing valid" });
+
+            try
+            {
+                //Checking the user
+                var user = await _userManager.CreateUser(userCreateInfo);
+                //Checking the homegroup
+                var home = await _homeManager.CreateHome(new HomeCreate() { Name = "Default" }, user.Id);
```

Comment:
why not take name as parameter and new HomeCreate in the method?

Reply:
Under UserCreate, snakkede vi om altid at være i en Default gruppe, så det er min begrundelse for dette.

Reply:
i explained it poorly, here is what i meant

class HomeManager
{
    Home CreateHome(string name, int userID)
   {
      var newHome = new HomeCreate() { Name = name };
    }
}

Reply:
Det er også basiclly det der bliver gjort i HomeManager.CreateHome()
Bare med HomeCreate Objectet, for at kunne skallerer objected hvis nødvendigt.
Og så er det consistent med de andre CreationControllers